### PR TITLE
Update plugins

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Update nearly all plugins listed by
```mvn versions:display-plugin-updates -nsu -B -Dversions.outputLineWidth=120 | grep -- '->' | sort -u```

* set minimum Maven version to 3.6.3
* build-helper-maven-plugin, 3.4.0 to 3.5.0, see https://github.com/mojohaus/build-helper-maven-plugin/releases/tag/3.5.0
* jococo-maven-plugin, 0.8.8 to 0.8.11, see https://www.jacoco.org/jacoco/trunk/doc/changes.html
* jvnet maven-jaxb2-plugin, 0.14.0 to 0.15.3
* frontend-maven-plugin, 1.12.1 to 1.15.0, see https://github.com/eirslett/frontend-maven-plugin/blob/master/CHANGELOG.md
* license-maven-plugin, 4.2 to 4.3
* maven-antrun-plugin, 1.8 to 3.1.0
* maven-clean-plugin, 3.3.1 to 3.3.2, see https://github.com/apache/maven-clean-plugin/releases
* maven-dependency-plugin, 3.6.0/3.1.2 to 3.6.1, see https://github.com/apache/maven-dependency-plugin/releases/tag/maven-dependency-plugin-3.6.1
* maven-enforcer-plugin, 3.4.0 to 3.4.1, see https://github.com/apache/maven-enforcer/releases/tag/enforcer-3.4.1
* maven-failsafe-plugin, 3.0.0-M7 to 3.2.5
* maven-jar-plugin, 3.3.0 to 3.4.1, see https://github.com/apache/maven-jar-plugin/releases
* maven-javadoc-plugin, 3.2.0 to 3.6.3, see https://github.com/apache/maven-javadoc-plugin/releases
* maven-plugin-plugin, 3.9.0 to 3.12.0, see https://github.com/apache/maven-plugin-tools/releases/tag/maven-plugin-tools-3.12.0
* maven-resources-plugin, 3.3.0 to 3.3.1
* maven-source-plugin, 3.3.0 to 3.3.1, see https://github.com/apache/maven-resources-plugin/releases/tag/maven-resources-plugin-3.3.1
* maven-surefire-plugin, 3.1.2 to 3.2.5, see https://github.com/apache/maven-surefire/releases
* spotless-maven-plugin, 2.38.0 to 2.43.0, see https://github.com/diffplug/spotless/blob/main/plugin-maven/CHANGES.md

* maven-shade-pluginm 3.5.2 to 3.5.3
* sortpom-maven-plugin, 3.3.0 to 3.4.1, see https://github.com/Ekryd/sortpom/releases
* openhab 18n-maven-plugin, 4.0.2 to 4.1.2

### CODE REVIEW

**Review by lsiepel — APPROVED** [review]
LGTM

Looks like most of the changes are minor and don't see any breaking changes or ones that i suspect to cause issues. I'll wait a few days for the codeowners to comment before i merge.

**Review by florian-h05 — APPROVED** [review]
LGTM for JS Scripting, thanks!

**Review by J-N-K — APPROVED** [review]

**Pull request merged**
